### DISCUSSION TITLE
[Feature] Advanced hide people

### DISCUSSION THREAD

**Discussion by Lokowitz** [immich-app/immich#10086] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

This is in addition to https://github.com/immich-app/immich/discussions/5480

It would be nice to have a more advanced option for hiding people. I want to define when and where the people should be hidden.

For example you have allot of fotos with people you don’t like or care about any more (ex-(girl-/boy-)friend etc.). For me it’s okay to have this fotos and I also don’t want to delete them but I don’t want to see them in my memories or places. And my grilfriend also don't want to see pictures of ey ex-girlfriend.

When:
- everytime
- In combination with known person or people
- Only this person is on the photo (and unknown people)

Where:
- everywhere
- People
- Timeline
- Places
- Memories
- Albums

## Explanations
- unknown people = recognized face without added a name
- known people = recognized face with added name

## Examples
Fotos should be hidden when the person or people and only the known selected people are on the foto. For example a selfie with my ex-girlfriend. Or another foto with my ex-girlfriend and other unknown people. 

But I still want to see a group foto with other known people.

### Photo 1 (hidden)
- me
- the person

### Photo 2 (hidden)
- me
- the person
- more or equal 0 unknown people

### Photo 3 (shown)
- me
- the person
- more then 0 known people
- more or equal 0 unknown people

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by Gorbush** [comment]
This 

> Photo 3 (shown)
> me
> the person
> more then 0 known people

conflicts with this:

> But I still want to see a group foto with other known people.

if your girlfriend is looking at the photo of you with "the person"

**Comment by rekcodocker** [comment]
Very specific for your usecase. Error-prone. And it will cost you more time than you think. 

The problem is that you have some photo's you want to _keep_ but not _show_ in a certain album.
There is a solution available:  Move them to another album.
 
Automating this and with all the rules (_this_ person but not _thas_ person except when _that_ person OR there are more than 3 knowns BUT no more than 20 unknowns AND it's not in a zoo) leads to unexpected results and costs a significant amount of testing.


The benefits are small, possibly negative.  
The first few rules will filter 80% of the photos you want.   Then you start tweaking to get to 100.  Tweak rule, test album, repeat.
That is going to lead to unexpected results, you are going to build more complicated than anyone has ever forseen, and it will lead to bugs. 

Immich has a method of organizing photos and it is albums, not a programmable rule-engine.

**Comment by bo0tzz** [comment]
Closing as a duplicate of the referenced request in combination with #1673
